File: ios/Classes/AliyunVoicePlugin.m
#import "AliyunVoicePlugin.h"
#import "Synthesizer.h"
#import "Aliyun.h"
#import "Recognizer.h"
@implementation AliyunVoicePlugin{
    Aliyun *aliyun;
    Recognizer *asr;
}
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"aliyun_voice"
            binaryMessenger:[registrar messenger]];
  AliyunVoicePlugin* instance = [[AliyunVoicePlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}
//@dynamic <#property#> Synthesizer* ;
//Synthesizer *aliyun;

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result{
    if(aliyun == NULL){
        aliyun=  [[Aliyun alloc] init];
    }
    if(asr == NULL){
        asr = [[Recognizer alloc] init];
    }
  if ([@"getPlatformVersion" isEqualToString:call.method]) {
    result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
  } else if ([@"initTTS" isEqualToString:call.method]) {
      [aliyun initTTS];
      [asr initASR];
      result(@"success");
  } else if ([@"speak" isEqualToString:call.method]) {
      [aliyun speak:call.arguments[@"content"]];
      //[aliyun startSynthesizer:call.arguments[@"content"]];
      result(@"success");
  }else if ([@"stop" isEqualToString:call.method]) {
     // [[Baidu CDSingletonBaidu] stop];
      [aliyun stop];
  }else if ([@"release" isEqualToString:call.method]) {
      [aliyun stop];
  }else if ([@"pauseOrResume" isEqualToString:call.method]) {
      [aliyun pauseOrResume:call.arguments[@"status"]];
      // [aliyun stop];
  }else if ([@"startASR" isEqualToString:call.method]) {
      [asr startRecognize];
  }else if ([@"stopASR" isEqualToString:call.method]) {
      [asr stopRecognize];
      // [aliyun stop];
  }
    

    
  else {
    result(FlutterMethodNotImplemented);
  }
}

@end

File: android/src/main/java/com/xsznet/aliyun_voice/AliyunVoicePlugin.java
package com.xsznet.aliyun_voice;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;

/** AliyunVoicePlugin */
public class AliyunVoicePlugin implements MethodCallHandler {
  protected static Registrar registrar;
  /** Plugin registration. */
  public static void registerWith(Registrar reg) {
    registrar = reg;
    final MethodChannel channel = new MethodChannel(registrar.messenger(), "aliyun_voice");
    channel.setMethodCallHandler(new AliyunVoicePlugin());
    BaiduSDK.getBaiduSDKIntance().attchActivity(registrar.activity());
  }

  @Override
  public void onMethodCall(MethodCall call, Result result) {
    if (call.method.equals("getPlatformVersion")) {
      result.success("Android " + android.os.Build.VERSION.RELEASE);
    } else   if (call.method.equals("initTTS")) {
      BaiduSDK.getBaiduSDKIntance().initTTs(registrar.activity());
    }else   if (call.method.equals("attach")) {
      BaiduSDK.getBaiduSDKIntance().attchActivity(registrar.activity());
    }    else   if (call.method.equals("speak")) {
      BaiduSDK.getBaiduSDKIntance().speak(call.argument("content").toString(),true,false,false,false);
    }else if(call.method.equals("release")){
      BaiduSDK.getBaiduSDKIntance().release();
    }
    else if(call.method.equals("stop")){
      BaiduSDK.getBaiduSDKIntance().stop();
    }
    else if(call.method.equals("pauseOrResume")){
      BaiduSDK.getBaiduSDKIntance().pauseOrResume(call.argument("status").equals("pause")?true:false);
    }
    else if(call.method.equals("startASR")){
      BaiduSDK.getBaiduSDKIntance().startASR();
    }
    else if(call.method.equals("stopASR")){
      BaiduSDK.getBaiduSDKIntance().stopASR();
    }
    else {
      result.notImplemented();
    }
  }
}
